### PR TITLE
Page to check call center UCR tables against ES

### PR DESCRIPTION
**Pull request by snopoke**
image

@benrudolph

### CODE REVIEW

File: corehq/apps/hqadmin/views.py [hunk 1]
```diff
@@ -831,3 +832,94 @@ def _get_submodules():
...
+            return {}
+
+        context = {
+            'domains': []
+        }
+        domain = self.request.GET.get('domain', None)
+        if domain:
+            domains = [domain]
+        else:
+            domains = [domain.name for domain in get_call_center_domains()]
```
Comment (dimagimon) [inline corehq/apps/hqadmin/views.py]: F812 list comprehension redefines 'domain' from line 861

File: corehq/apps/hqadmin/views.py [hunk 2]
```diff
@@ -831,3 +832,94 @@ def _get_submodules():
         line.strip()[1:].split()[1]
         for line in git.submodule()
     ]
+
+
+
+
+class CallcenterUCRCheck(BaseAdminSectionView):
```
Comment (dimagimon) [inline corehq/apps/hqadmin/views.py]: E303 too many blank lines (4)

File: corehq/apps/es/tests/test_aggregations.py [hunk 3]
```diff
@@ -370,3 +370,32 @@ def test_histogram_aggregation(self):
         res = ESQuerySet(example_response, query)
         output = res.aggregations.forms_by_date.raw_buckets
         self.assertEqual(output, expected_output)
+
+    def test_nested_aggregation(self):
```
Comment (dimagimon) [inline corehq/apps/es/tests/test_aggregations.py]: F811 redefinition of unused 'test_nested_aggregation' from line 29

**Comment by snopoke** [comment]
:white_check_mark: cc @czue 

would prefer if there was an automated alert for this but the performance cost of doing that regularly seemed high

**Comment by snopoke** [comment]
cc @esoergel

File: corehq/apps/es/aggregations.py [hunk 4]
```diff
@@ -455,12 +455,29 @@ def __init__(self, name, datefield, interval, timezone=None):
...
+
 AggregationTerm = namedtuple('AggregationTerm', ['name', 'field'])
 
 
-class NestedAggregationHelper(object):
+class NestedTermAggregationsHelper(object):
     """
-    Helper to run nested term-based queries (equivalent to SQL group-by clauses)
+    Helper to run nested term-based queries (equivalent to SQL group-by clauses).
+    This is not at all related to the ES 'nested aggregation'.
```
Comment (czue) [inline corehq/apps/es/aggregations.py]: well that was a confusing name i chose. :disappointed:

**Comment by czue** [comment]
this is awesome. now we should just build the same thing for primary forms/cases --> ES. :smiling_imp:

File: corehq/apps/es/aggregations.py (hunk 4 above)
Comment (snopoke) [inline corehq/apps/es/aggregations.py]: I spend a while trying to come up with a better name but couldn't think of much else. `SubAggregationHelper` just didn't seem right.

File: corehq/apps/es/tests/test_aggregations.py [hunk 5]
```diff
@@ -370,3 +370,32 @@ def test_histogram_aggregation(self):
...
+            },
+            "size": SIZE_LIMIT
+        }
+        query = HQESQuery('cases').aggregation(
+            NestedAggregation(
+                'case_actions',
+                'actions',
+            )
+        )
+        self.checkQuery(query, json_output)
```
Comment (esoergel) [inline corehq/apps/es/tests/test_aggregations.py]: :star:

**Pull request merged**